packages/evm/src/chainMethods/transferNative.ts

```diff
@@ -21,8 +22,8 @@ export const makeTransferNative =
   async (options: TransferNativeOptions) => {
     const to = EvmAddress.create(options.to);
     const value = EvmNative.create(options.value);
-    // TODO: check if provided chain is same as current chain, otherwise throw error
     const chain = assertChain(_chain, 'Chain is not set on MoralisEvm. Make sure to be properly connected');
+    if (options.chain) validateChain(options.chain, chain);
```

Comment:
Can you add curly braces? that's easier to read.

I will add an eslint rule for this in a separate PR

```suggestion
    if (options.chain) {
      validateChain(options.chain, chain);
    }
```

Reply:
Same for the other files